frontend/src/main/scala/bloop/cli/Proxy.scala

```diff
@@ -0,0 +1,90 @@
+package bloop.cli
+
+import java.net.URL
+import java.net.MalformedURLException
+
+import scala.util.control.NonFatal
+
+import bloop.engine.State
+
+/**
+  * Proxy helper module
+  * Information came from  https://docs.oracle.com/javase/8/docs/technotes/guides/net/proxies.html
+  */
+object Proxy {
+  /**
+    * Extract environments variables from the given state to set the proxy properties of the jvm.
+    * If environement variable are not found in the state, the property corresponding is cleared,
+    * to let user add and remove env var.
+    * */
+  def setProxyJvmSettings(state : State) : Unit = {
+    //no_proxy environment variable seems not be binded to a protocol.
+
+    val maybeNoProxy = Option(state.commonOptions.env.getProperty("no_proxy")).map(_.replace(',','|'))
+
+    val maybeHttpProxy = state.commonOptions.env.getProperty("http_proxy")
+    if(maybeHttpProxy != null) {
+      try {
+        val url =  new URL(maybeHttpProxy)
+        System.setProperty("http.proxyHost",url.getHost())
+        System.setProperty("http.proxyPort",url.getPort().toString)
+        maybeNoProxy.foreach(System.setProperty("http.nonProxyHosts", _))
+      } catch {
+        case e : MalformedURLException => //Ignore, but should maybe log a warning if the URL is not parsable.
+      }
+    } else {
+      System.clearProperty("http.proxyHost")
+      System.clearProperty("http.proxyPort")
+      if(maybeNoProxy.isEmpty)
+        System.clearProperty("http.nonProxyHosts")
+    }
+
+    val maybeHttpsProxy = state.commonOptions.env.getProperty("https_proxy")
+    if(maybeHttpsProxy != null) {
```

Comment:
Can we reformat all of this file with `scalafmt`? The style scalafmt file is in the repository so it will pick up the changes.

Reply:
That not down automatically when compiling?

Reply:
No, I think it's better to keep both separate, check https://scalameta.org/scalafmt/ if you need more info to get set up